src/ui/map.js

```diff
@@ -1061,7 +1061,7 @@ class Map extends Camera {
      * | [`sourcedataloading`](#map.event:sourcedataloading)       |                           |
      * | [`styleimagemissing`](#map.event:styleimagemissing)       |                           |
      *
-     * @param {string} layerId (optional) The ID of a style layer. If you provide a `layerId`,
+     * @param {string} [layerId] (optional) The ID of a style layer. If you provide a `layerId`,
```

Comment:
Is there a case where square brackets would be present but not "(optional)" or vice versa?

At the moment this seems to duplicate information in a way which could _probably be enforced with linting_ 😉, but even better might be not to duplicate information in the first place if they are indeed 1:1 identical.

Reply:
Good questions and I like where you're going with this @rreusser ! 

I think that anywhere `(optional)` appears, the term preceding it should be in square brackets, but I am not 100% sure. 

Looking in the [source code for `Evented`](https://github.com/mapbox/mapbox-gl-js/blob/c377a90e768c48b7a422596740e06cae2a55a055/src/util/evented.js#L55-L183) I see a `?` on line 100 which makes me think that that is the most reliable indication that a parameter is optional. I would love a more certain recommendation from the product team to enable more reliable and future-proof linting.

Reply:
If I understand correctly, it seems like we have three ways to indicate optional arguments:

1. square brackets
2. (optional)
3. Flow/TypeScript annotations

My concern here is that I think maybe 2 should be eliminated, deemed exactly equivalent to 1, and inserted programatically, but I think 3 is a slightly different thing which is reasonable to maintain separately.

@ryanhamley makes a good point that semantics of flow/typescript optional parameters (via question marks) differ from what we want to communicate in the docs. For example, if a middle argument is optional, TypeScript will respond with: `A required parameter cannot follow an optional parameter.`

There is a [check-params-names](https://github.com/gajus/eslint-plugin-jsdoc#eslint-plugin-jsdoc-rules-check-param-names) eslint-jsdoc rule that matches up arguments and could probably be extended to apply meaningful logic with enough tolerance to accommodate these differences, but this may be overkill. (like if a middle argument is jsdoc-optional, then we could enforce the next closest type annotations which match that).